eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/DatasetAlleleResponse.java

```diff
@@ -33,6 +36,21 @@
     String externalUrl;
     HashMap<String,String> info;
 
+    public static DatasetAlleleResponse buildDatasetAlleleResponse(boolean exists, VariantSourceMongo variantSourceMongo, Map<String,Float> studyIdToFrequencyMapper) {
```

Comment:
- I think this method doesn't need the whole frequency mapper, only the frequency
- After seeing the actual factory method, I now think that this is a too specific use case, and it should not be in this bean. I think it fits better as a helper method in GA4GHBeaconWSServerV2.java, right next to where it's used. This would make things more clean to keep these models independent

Reply:
fixed :)